### PR TITLE
[Improve][SourceConnector] Unified schema parameter, update IoTDB sou…

### LINKED ISSUE #3823: [Improve][SourceConnector] Unified schema parameter, update IoTDB source fields to schema
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

[Improve][SourceConnector] Unified schema parameter, update IoTDB source `fields` to `schema`

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sunnyzhuzhu**
## Purpose of this pull request
close https://github.com/apache/incubator-seatunnel/issues/3823

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by CalvinKirs — COMMENTED** [review]
TBD: doc

**Comment by sunnyzhuzhu** [comment]
> TBD: doc

Thanks for your suggestions！I have updated doc and checked code style with docs/contribution/coding-guide.md. Can you help to approve the ci workflows again, since this is my first-time contribution. Thanks very much!  @CalvinKirs

**Comment by TyrantLucifer** [comment]
Thank you for your contribution. Let's waiting CI process.

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java [hunk 1]
```diff
@@ -77,8 +77,13 @@ public void prepare(Config pluginConfig) throws PrepareFailException {
                 );
             }
         }
-        SeaTunnelSchema seatunnelSchema = SeaTunnelSchema.buildWithConfig(pluginConfig);
-        this.typeInfo = seatunnelSchema.getSeaTunnelRowType();
+        if (pluginConfig.hasPath(SeaTunnelSchema.SCHEMA.key())) {
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java]: Schema is a required key of connector, so if the user does not configured it connector should throw exception. You can refer to #3897 . Thanks!

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java [hunk 2]
```diff
@@ -77,8 +77,13 @@ public void prepare(Config pluginConfig) throws PrepareFailException {
...
+            Config schemaConfig = pluginConfig.getConfig(SeaTunnelSchema.SCHEMA.key());
+            this.typeInfo = SeaTunnelSchema.buildWithConfig(schemaConfig).getSeaTunnelRowType();
+        } else {
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java]: Remove else branch.

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]
Please update some code as #3897. You should update option rulle

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java [hunk 3]
```diff
@@ -65,21 +65,29 @@ public String getPluginName() {
...
+        try {
+            Config schemaConfig = pluginConfig.getConfig(SeaTunnelSchema.SCHEMA.key());
+            this.typeInfo = SeaTunnelSchema.buildWithConfig(schemaConfig).getSeaTunnelRowType();
+            pluginConfig.entrySet().forEach(entry -> configParams.put(entry.getKey(), entry.getValue().unwrapped()));
+        } catch (Exception e) {
+            throw new IotdbConnectorException(SeaTunnelAPIErrorCode.CONFIG_VALIDATION_FAILED,
+                String.format("PluginName: %s, PluginType: %s, Message: %s",
+                    getPluginName(), PluginType.SOURCE, e)
+            );
         }
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java]:
```suggestion
        SeaTunnelSchema seaTunnelSchema = SeaTunnelSchema.buildWithConfig(pluginConfig.getConfig(SeaTunnelSchema.SCHEMA.key()));
        this.typeInfo = seaTunnelSchema.getSeaTunnelRowType();
        pluginConfig.entrySet().forEach(entry -> configParams.put(entry.getKey(), entry.getValue().unwrapped()));
```

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java]: Revert this logic. As is.

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSourceFactory.java [hunk 4]
```diff
@@ -49,7 +50,7 @@ public String factoryIdentifier() {
     @Override
     public OptionRule optionRule() {
         return OptionRule.builder()
-                .required(NODE_URLS, USERNAME, PASSWORD, SQL)
+                .required(NODE_URLS, USERNAME, PASSWORD, SQL, SeaTunnelSchema.SCHEMA)
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSourceFactory.java]: static import is better

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java (hunk 3 above)
Comment (sunnyzhuzhu) [reply seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java]: @TyrantLucifer Thanks for your suggestions.I have done this.

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSourceFactory.java (hunk 4 above)
Comment (sunnyzhuzhu) [reply seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSourceFactory.java]: @TyrantLucifer Thanks for your suggestions.I have done this.

**Review by TyrantLucifer — APPROVED** [review]
LGTM, thank you for your contirbution. Looking forward your next pull request! BTW, before this pr be merged CI should be passed. Let's waiting CI. cc @hailin0 @Hisoka-X @EricJoy2048 @CalvinKirs

**Comment by CalvinKirs** [comment]
> LGTM, thank you for your contirbution. Looking forward your next pull request! BTW, before this pr be merged CI should be passed. Let's waiting CI. cc @hailin0 @Hisoka-X @EricJoy2048 @CalvinKirs↳

I have one question, like this https://github.com/apache/incubator-seatunnel/pull/3959

**Comment by TyrantLucifer** [comment]
> > LGTM, thank you for your contirbution. Looking forward your next pull request! BTW, before this pr be merged CI should be passed. Let's waiting CI. cc @hailin0 @Hisoka-X @EricJoy2048 @CalvinKirs↳
> 
> I have one question, like this #3959

Unified parameters is convenient for frontend display the parameters. Do I understand right? @EricJoy2048

**Review by liugddx — APPROVED** [review]

**Comment by EricJoy2048** [comment]
> > > LGTM, thank you for your contirbution. Looking forward your next pull request! BTW, before this pr be merged CI should be passed. Let's waiting CI. cc @hailin0 @Hisoka-X @EricJoy2048 @CalvinKirs↳
> > 
> > 
> > I have one question, like this #3959
> 
> Unified parameters is convenient for frontend display the parameters. Do I understand right? @EricJoy2048

Yes, I like this pr. Thanks for your contirbution.

**Comment by CalvinKirs** [comment]
> > > > LGTM, thank you for your contirbution. Looking forward your next pull request! BTW, before this pr be merged CI should be passed. Let's waiting CI. cc @hailin0 @Hisoka-X @EricJoy2048 @CalvinKirs↳↳
> > > 
> > > 
> > > I have one question, like this #3959↳
> > 
> > 
> > Unified parameters is convenient for frontend display the parameters. Do I understand right? @EricJoy2048↳
> 
> Yes, I like this pr. Thanks for your contirbution.↳

I think you didn't get my point, what we need to consider is compatibility

**Comment by wfrong** [comment]
> > > > > LGTM, thank you for your contirbution. Looking forward your next pull request! BTW, before this pr be merged CI should be passed. Let's waiting CI. cc @hailin0 @Hisoka-X @EricJoy2048 @CalvinKirs↳↳
> > > > 
> > > > 
> > > > I have one question, like this #3959↳
> > > 
> > > 
> > > Unified parameters is convenient for frontend display the parameters. Do I understand right? @EricJoy2048↳
> > 
> > 
> > Yes, I like this pr. Thanks for your contirbution.↳
> 
> I think you didn't get my point, what we need to consider is compatibility

I think you mean we should support  previous usage only config fields without schema?

**Comment by sunnyzhuzhu** [comment]
> > > > > LGTM, thank you for your contirbution. Looking forward your next pull request! BTW, before this pr be merged CI should be passed. Let's waiting CI. cc @hailin0 @Hisoka-X @EricJoy2048 @CalvinKirs↳↳
> > > > 
> > > > 
> > > > I have one question, like this #3959↳
> > > 
> > > 
> > > Unified parameters is convenient for frontend display the parameters. Do I understand right? @EricJoy2048↳
> > 
> > 
> > Yes, I like this pr. Thanks for your contirbution.↳
> 
> I think you didn't get my point, what we need to consider is compatibility

Sounds reasonable! Should I change on this pr to support previous usage that may exists in some users' jobs configure? @TyrantLucifer @EricJoy2048 @CalvinKirs

**Pull request merged**